src/core/sys/posix/pthread.d

```diff
@@ -499,8 +499,11 @@ int pthread_key_delete(pthread_key_t);
 int pthread_mutex_destroy(pthread_mutex_t*);
 int pthread_mutex_init(pthread_mutex_t*, pthread_mutexattr_t*) @trusted;
 int pthread_mutex_lock(pthread_mutex_t*);
+int pthread_mutex_lock(shared(pthread_mutex_t)*);
```

Comment:
Oh yeah, if I remember correctly that was changed recently. The correct way to go about in LDC, in my opinion, is to allow overloading if only if all overloads have the same ABI as should be the case here.

To be pedantically correctly, those functions should have been declared as taking `shared` parameters in the first place, but leaving only the `shared` versions would be a breaking change now.